src/model/image-tag.ts

```diff
@@ -35,25 +35,61 @@ class ImageTag {
       webgl: 'webgl',
       mac: 'mac-mono',
       windows: 'windows-mono',
+      windowsIl2cpp: 'windows-il2cpp',
+      wsaplayer: 'universal-windows-platform',
       linux: 'base',
       linuxIl2cpp: 'linux-il2cpp',
       android: 'android',
       ios: 'ios',
+      tvos: 'appletv',
       facebook: 'facebook',
     };
   }
 
   static getTargetPlatformToImageSuffixMap(platform, version) {
-    const { generic, webgl, mac, windows, linux, linuxIl2cpp, android, ios, facebook } = ImageTag.imageSuffixes;
+    const {
+      generic,
+      webgl,
+      mac,
+      windows,
+      windowsIl2cpp,
+      wsaplayer,
+      linux,
+      linuxIl2cpp,
+      android,
+      ios,
+      tvos,
+      facebook,
+    } = ImageTag.imageSuffixes;
 
     const [major, minor] = version.split('.').map((digit) => Number(digit));
     // @see: https://docs.unity3d.com/ScriptReference/BuildTarget.html
     switch (platform) {
       case Platform.types.StandaloneOSX:
         return mac;
       case Platform.types.StandaloneWindows:
+        // Can only build windows-il2cpp on a windows based system
+        if (process.platform === 'win32') {
+          // Unity versions before 2019.3 do not support il2cpp
+          if (major >= 2020 || (major === 2019 && minor >= 3)) {
+            return windowsIl2cpp;
+          } else {
+            throw new Error(`Windows-based builds are only supported on 2019.3.X+ versions of Unity.
+                             If you are trying to build for windows-mono, please use a Linux based OS.`);
+          }
+        }
         return windows;
       case Platform.types.StandaloneWindows64:
+        // Can only build windows-il2cpp on a windows based system
+        if (process.platform === 'win32') {
+          // Unity versions before 2019.3 do not support il2cpp
+          if (major >= 2020 || (major === 2019 && minor >= 3)) {
+            return windowsIl2cpp;
+          } else {
+            throw new Error(`Windows-based builds are only supported on 2019.3.X+ versions of Unity.
+                             If you are trying to build for windows-mono, please use a Linux based OS.`);
+          }
+        }
```

Comment:
This looks great for the platform selection part. Love it!

Reply:
Since this code is duplicated for both the `Platform.types.StandaloneWindows` and `Platform.types.StandaloneWindows64` cases, what do you think of simply doing a fall through? Some people don't like fall throughs (some people think switch statements should always be avoided), but I think this is the exact situation where a switch fall through is good since it removes duplicate code:
```suggestion
      case Platform.types.StandaloneWindows64:
        // Can only build windows-il2cpp on a windows based system
        if (process.platform === 'win32') {
          // Unity versions before 2019.3 do not support il2cpp
          if (major >= 2020 || (major === 2019 && minor >= 3)) {
            return windowsIl2cpp;
          } else {
            throw new Error(`Windows-based builds are only supported on 2019.3.X+ versions of Unity.
                             If you are trying to build for windows-mono, please use a Linux based OS.`);
          }
        }
```